sepal_ui/mapping/mapping.py

```diff
@@ -785,21 +689,125 @@ def get_viz_params(image):
 
+        layer (ipyleaflet.Layer): any layer type from ipyleaflet
         hover (bool): whether to use the default hover style or not.
+        """
+
+        # remove existing layer before addition
+        existing_layer = self.find_layer(layer.name)
+        not existing_layer or self.remove_layer(existing_layer)
 
+        # apply default coloring for geoJson
+        if isinstance(layer, ipl.GeoJSON):
+            layer.style = layer.style or styles.layer_style
+            hover_style = styles.layer_hover_style if hover else layer.hover_style
+            layer.hover_style = layer.hover_style or hover_style
+
+        super().add_layer(layer)
+
+        return
+
+    def add_basemap(self, basemap="HYBRID"):
         """
+        Adds a basemap to the map.
 
-        if isinstance(layer, GeoJSON):
+        Args:
+            basemap (str, optional): Can be one of string from basemaps. Defaults to 'HYBRID'.
+        """
+        if basemap not in basemaps.keys():
+            keys = "\n".join(basemaps.keys())
+            msg = f"Basemap can only be one of the following:\n{keys}"
+            raise ValueError(msg)
 
-            if not layer.style:
-                layer.style = styles.layer_style
+        self.add_layer(basemaps[basemap])
 
-            if hover and not layer.hover_style:
-                layer.hover_style = styles.layer_hover_style
+        return
 
-        super().add_layer(layer)
+    def get_scale(self):
+        """
+        Returns the approximate pixel scale of the current map view, in meters.
+        Reference: https://blogs.bing.com/maps/2006/02/25/map-control-zoom-levels-gt-resolution
 
-        return self
+        Returns:
+            (float): Map resolution in meters.
+        """
+
+        return 156543.04 * math.cos(0) / math.pow(2, self.zoom)
+
+    def find_layer(self, key):
+        """
+        Search a layer by name or index
+
+        Args:
+            key (Layer, str, int): the layer name, index or directly the layer
+
+        Return:
+            (TileLLayerayer): the first layer using the same name or index else None
+        """
+
+        if isinstance(key, str):
+            layer = next((lyr for lyr in self.layers if lyr.name == key), None)
+        elif isinstance(key, int):
+            size = len(self.layers)
+            layer = self.layers[key] if -size <= key < size else None
+        elif isinstance(key, ipl.Layer):
+            layer = next((lyr for lyr in self.layers if lyr == key), None)
+        else:
+            raise ValueError(f"key must be a int or a str, {type(key)} given")
+
+        return layer
+
+    # ##########################################################################
+    # ###                overwrite geemap calls                              ###
+    # ##########################################################################
+
+    setCenter = set_center
+    centerObject = zoom_ee_object
+    addLayer = add_ee_Layer
```

Comment:
apart from remaining Pythonic... no. So that's only my OTC speaking.
Also That was already the case in `geemap` so I have some modules that are using the Python names instead of the GEE one

Reply:
Ok. I get then the use, just define in pythonic way, but make them available in its "ee API" style.